toolkit/tools/imagegen/attendedinstaller/primitives/enumfield/enumfield.go

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package enumfield
+
+import (
+	"math"
+
+	"github.com/gdamore/tcell"
+	"github.com/rivo/tview"
+)
+
+type EnumField struct {
+	*tview.Box
+
+	// The label of the field
+	label string
+
+	// Maximum width of the label, or 0 for label's length
+	labelWidth int
+
+	// Options to choose from
+	options []string
+
+	// Index of the currently selected option
+	selectedOption int
+
+	// The label color.
+	labelColor tcell.Color
+
+	// The label color when this item is selected.
+	labelColorActivated tcell.Color
+
+	// The text color.
+	textColor tcell.Color
+
+	// The text color when this item is selected.
+	textColorActivated tcell.Color
+
+	// The background color.
+	backgroundColor tcell.Color
+
+	// The background color when this item is selected.
+	backgroundColorActivated tcell.Color
+
+	// A callback invoked when the user leaves this form item
+	onFinished func(tcell.Key)
+
+	// A callback invoked when this primitive receives focus.
+	onFocus func()
+
+	// A callback invoked when this primitive loses focus.
+	onBlur func()
+}
+
+// NewEnumField returns a new navigation bar.
+func NewEnumField(options []string) *EnumField {
+	return &EnumField{
+		Box:                      tview.NewBox(),
+		backgroundColor:          tview.Styles.PrimitiveBackgroundColor,
+		backgroundColorActivated: tview.Styles.ContrastBackgroundColor,
+		labelColor:               tview.Styles.SecondaryTextColor,
+		labelColorActivated:      tview.Styles.ContrastSecondaryTextColor,
+		textColor:                tview.Styles.PrimaryTextColor,
+		textColorActivated:       tview.Styles.PrimaryTextColor,
+		options:                  options,
```

Comment:
Should we check if `options` is not `nil` at this point? It seems we're assuming that we're working with a valid slice in the rest of this code.

Reply:
I've added a check for drawing, so we don't draw an empty option list. However,  mimicking the behavior of other primitives, I don't check for nil.

Reply:
As long as we don't break anywhere trying to dereference `nil`.